### DISCUSSION TITLE
Merge multiple files together (with array append)

### DISCUSSION THREAD

**Discussion by coryroloff** [mikefarah/yq#737] [Q&A] [answered]
Hello!

Is there a way to merge multiple files together while telling it to append arrays together without having to specify all of the files in the query? For example, there's "select(fileIndex == 0) *+ select(fileIndex == 1)" but for that to work you have to add the select(fileIndex) for each file. Is there a more generic query that will just look at the list of files you give it and automatically merge them all together while appending arrays?

**Comment by mikefarah** [comment, marked as answer]
https://mikefarah.gitbook.io/yq/usage/tips-and-tricks#merge-combine-all-documents-into-one

**Reply by mikefarah** [reply]
You'll need to `*+` instead of just `*`

**Reply by coryroloff** [reply]
Thanks!
